distributed/client.py

```diff
@@ -1473,6 +1479,28 @@ def _handle_error(self, exception=None):
         logger.warning("Scheduler exception:")
         logger.exception(exception)
 
+    async def _story(self, keys=(), on_error="raise"):
+        stimulus_id = f"client-story-{time()}"
+        task = asyncio.ensure_future(
+            self.scheduler.cluster_story(
+                keys=keys, on_error=on_error, stimulus_id=stimulus_id
+            )
+        )
+
+        try:
+            return await task
+        except Exception:
+            if on_error == "raise":
+                task.cancel()
```

Comment:
the task will always be done at this point - because of the `return await task(`